### DISCUSSION TITLE
Change to then when method

### DISCUSSION THREAD

**Discussion by Squad-G** [laravel/framework#44230] [Ideas]
`Hello,

I don't know really how to submit new ideas so I'll be doing it here.

I know Laravel has started to replace a lot of code to use the new PHP8 keyword `match`.

We've started doing the same in some of our projects and we really like this piece of code:

```
match(true)
{
  true => "it's true !!",
  false => "sad face"
};
```

So I would like to submit the following idea, to replace the `when` method of the trait `ConditionallyLoadsAttributes`.

Original:

```
protected function when($condition, $value, $default = null)
{
	if ($condition) {
		return value($value);
	}

	return func_num_args() === 3 ? value($default) : new MissingValue;
}
```

New:

```
protected function when($condition, $value, $default = null)
{
	return match($condition)
	{
		true => value($value),
		false => func_num_args() === 3 ? value($default) : new MissingValue
	};
}
```

To me, while it takes the same amount of line, I find it easier to read.
This may be a bad example since it has 4-5 lines but in a class with a lot of lines, this is easier to spot.

Thank you :)

**Comment by juse-less** [comment]
This would actually cause unintended side effects, and be a breaking change to the current API.
The `match` language construct does a strict comparison, unlike `switch`, expressions used in `if`'s, ternaries, etc.
So if it contains a falsey value like null, an empty string, or 0, it wouldn't match `false`.
Same thing if you provide a truey value like a non-empty string, it won't match `true`.

**Reply by Squad-G** [reply]
You are right.

If `$condition = null;`, then it will not match any statement.

I just never thought to use this method with a value of `null` for example ... but it could break alot of stuff indeed.
